### PR TITLE
Make assigning segments faster

### PR DESCRIPTION
**Pull request by yah01**
This improve the Load performance,
and let the LoadBalance fails fast, which allows us to retry it timely

Signed-off-by: yah01 <yang.cen@zilliz.com>
#17321 #17016

### CODE REVIEW

**Comment by yah01** [comment]
rerun ut

File: internal/querycoord/querynode.go [hunk 1]
```diff
@@ -335,7 +336,10 @@ func (qn *queryNode) getNodeInfo() (Node, error) {
 	if err != nil {
 		return nil, err
 	}
-	resp, err := qn.client.GetMetrics(qn.ctx, req)
+
+	ctx, cancel := context.WithTimeout(qn.ctx, time.Second)
```
Comment (congqixia) [inline internal/querycoord/querynode.go:340]: NIT: magic number for timeout duration. Maybe put it into configuration or const in the future.

File: internal/querycoord/segment_allocator.go [hunk 2]
```diff
@@ -227,3 +215,23 @@ func nodeIncluded(nodeID int64, includeNodeIDs []int64) bool {
 
 	return false
 }
+
+func nodesFilter(nodes []UniqueID, include []UniqueID, exclude []UniqueID) []UniqueID {
+	result := make([]UniqueID, 0)
```
Comment (congqixia) [inline internal/querycoord/segment_allocator.go:220]: NIT: please do not use 0 length allocation

**Review by congqixia — APPROVED** [review]
/lgtm

**Pull request merged**